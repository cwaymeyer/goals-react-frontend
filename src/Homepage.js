import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import UserContext from "./context/userContext";
import { Container, Button } from "@mui/material";

function Homepage({ logout }) {
  const navigate = useNavigate();
  const { currentUser } = useContext(UserContext);

  function handleLogout() {
    logout();
    navigate("/login");
  }

  return (
    <Container maxWidth="sm">
      <h1>Hello!</h1>
      <Button
        variant="contained"
        size="small"
        color="warning"
        onClick={handleLogout}
      >
        Log out
      </Button>
    </Container>
  );
}

export default Homepage;
