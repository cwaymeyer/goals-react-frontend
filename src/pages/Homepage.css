body {
  background-color: rgb(244, 248, 252);
}

.Homepage {
  background-color: rgb(244, 248, 252);
}

.Homepage .create-goal button {
  background-color: rgb(255, 255, 255);
}

.Goal a {
  text-align: center;
  text-decoration: none;
}
