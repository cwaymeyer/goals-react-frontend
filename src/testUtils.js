import UserContext from "./context/userContext";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const progress = [
  {
    id: 10,
    goal_id: 1,
    weight: 50,
    reps: 5,
    orm: "56.3",
    date: "1645477469350",
  },
  {
    id: 11,
    goal_id: 1,
    weight: 50,
    reps: 6,
    orm: "58.1",
    date: "1645477669350",
  },
];

const starting_progress = {
  orm: "56.3",
  date: "1645477269350",
};

const latest_progress = {
  orm: "58.1",
  date: "1645477669350",
};

const goal = {
  id: "1",
  name: "Test Goal",
  username: "demoUser",
  start_weight: null,
  target_weight: 100,
  timeline: 3,
  start_date: "1645477269350",
  end_date: "1653163269350",
  progress: progress,
  starting_progress: starting_progress,
  latest_progress: latest_progress,
};

const demoUser = {
  username: "demoUser",
  password: "password",
  goals: [goal],
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

const renderWithRouter = (ui, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);

  return render(ui, { wrapper: BrowserRouter });
};

export {
  UserProvider,
  renderWithRouter,
  goal,
  starting_progress,
  latest_progress,
};
