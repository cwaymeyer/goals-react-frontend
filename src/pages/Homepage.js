import "./Homepage.css";
import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import UserContext from "../context/userContext";
import { Container, Button, Grid } from "@mui/material";
import HomeLogo from "../HomeLogo";
import AddCircleIcon from "@mui/icons-material/AddCircle";
import GoalList from "../GoalList";

function Homepage({ logout }) {
  const navigate = useNavigate();
  const { currentUser } = useContext(UserContext);

  function handleLogout() {
    logout();
    navigate("/login");
  }

  return (
    <div className="Homepage">
      <HomeLogo />
      <Container maxWidth="md">
        <Grid container>
          <Button
            variant="contained"
            size="small"
            color="warning"
            onClick={handleLogout}
            sx={{ position: "absolute", right: 0, mr: 5, mt: 5 }}
          >
            Log out
          </Button>
          <Grid item xs={12} fontSize={30}>
            <h1>Hello, {currentUser.username}</h1>
          </Grid>
          <Grid item xs={12}>
            <div className="create-goal">
              <Button
                variant="outlined"
                size="large"
                startIcon={<AddCircleIcon />}
                onClick={() => navigate("/create")}
              >
                Create new goal
              </Button>
            </div>
          </Grid>
          <GoalList />
        </Grid>
      </Container>
    </div>
  );
}

export default Homepage;
