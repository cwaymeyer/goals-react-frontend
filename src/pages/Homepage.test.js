import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("displays correctly", function () {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(getByText("Hello, demouser")).toBeInTheDocument();
  expect(getByText("Create new goal")).toBeInTheDocument();
});
