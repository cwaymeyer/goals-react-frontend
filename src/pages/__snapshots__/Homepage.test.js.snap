// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-c1fbe3-MuiSvgIcon-root"
        data-testid="FitnessCenterIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.57 14.86 22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
        />
      </svg>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedWarning MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root  css-1xi8xg6-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Log out
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1wbziol-MuiGrid-root"
        >
          <h1>
            Hello, demouser
          </h1>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="create-goal"
          >
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButtonBase-root  css-9y1egq-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AddCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
                  />
                </svg>
              </span>
              Create new goal
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-aqz1n5-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
          >
            <div
              class="Goal"
            >
              <a
                href="/1"
              >
                <div
                  class="paper"
                >
                  <div
                    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1a1n6j7-MuiPaper-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ehlm6r-MuiGrid-root"
                      >
                        <h3>
                          Test Goal
                        </h3>
                      </div>
                    </div>
                  </div>
                </div>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
